### DISCUSSION TITLE
TRPC error always propagates as HTTP 200

### DISCUSSION THREAD

**Discussion by ejabu** [trpc/trpc#6324] [General]
Why ?

Did I miss some environment variables ?

I just cloned https://github.com/trpc/examples-next-prisma-starter

and does not know why it produced error differently.

Please see below for details

## Production

![image](https://github.com/user-attachments/assets/b2c60c91-2864-4c65-9707-72a2233e46df)
- When form failed on zod validation, it throws 400

## Local Dev Server

![image](https://github.com/user-attachments/assets/942eb3e3-abe3-429c-b4a3-7e43bb584779)
- When form failed on zod validation, it throws 200

**Comment by KATT** [comment]
https://trpc.io/docs/rpc#error-response

**Reply by ejabu** [reply]
Thanks for ⚡ response. 

Unfortunately I still need your help on this.

<img width="1489" alt="image" src="https://github.com/user-attachments/assets/24b4e13e-32c7-4704-830f-3259a281afb1" />

I throw the error explicitly as the docs suggested, but I still got 200 in my network panel.

**Reply by KATT** [reply]
Open an issue if you think it's a bug, use https://github.com/trpc/examples-next-minimal-starter as a base for reproduction
